Pcap++/src/PcapFilter.cpp

```diff
@@ -348,11 +338,19 @@ void OrFilter::parseToString(std::string& result)
 		result += '(' + innerFilter + ')';
 		if (m_FilterList.back() != *it)
 		{
-			result += " or ";
+			result += m_CompositeDelimiter;
 		}
 	}
 }
 
+AndFilter::AndFilter() : CompositeFilter(" and ") {}
+
+AndFilter::AndFilter(std::vector<GeneralFilter*>& filters) : CompositeFilter(filters, " and ") {}
+
+OrFilter::OrFilter() : CompositeFilter(" or ") {}
```

Comment:
I don't know this part of the code so maybe a stupid remark, but if the only change is this string why not using a Template?

```
enum FilterOPs { OR, AND, XOR, NOT };

Template <FilterOPs operator>
class CompositeFilter : public GeneralFilter

using OrFilter = CompositeFilter<OR>
```

Reply:
I also like the way to use template.
note that a template function is required to get the correct delimiter according to the op type.

Reply:
Ok let's be consistent then, fine for me.

Reply:
I did play around with the templated version. It actually did not turn out too bad.
Gonna push it to a different branch on my fork and link it here if you want to check it when I'm done.

Reply:
This is the potential solution I came up with as a templated version: https://github.com/Dimi1010/PcapPlusPlus/pull/1

Reply:
Implemented in 2b3bb92 and polished a bit in the following commits.